ptyprocess/ptyprocess.py

```diff
@@ -289,10 +302,14 @@ def spawn(
                     os.execv(command, argv)
                 else:
                     os.execvpe(command, argv, env)
-            except OSError as err:
+            except Exception as err:
                 # [issue #119] 5. If exec fails, the child writes the error
                 # code back to the parent using the pipe, then exits.
-                tosend = 'OSError:{}:{}'.format(err.errno, str(err))
+                if isinstance(err, OSError):
+                    tosend = 'OSError:{}:{}'.format(err.errno, str(err))
+                else:
+                    cls_name = err.__class__.__name__
+                    tosend = 'Exception:0:{}: {}'.format(cls_name, str(err))
```

Comment:
Following the exception handling around `preexec_fn()`, maybe this should do `'{}:0:{}'` (e.g. `'UnicodeEncodeError:0:blah'`.